GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/Store/CarStorePageController.cs

```diff
@@ -0,0 +1,126 @@
+using System.Globalization;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+// controller for car store page
+public class CarStorePageController : MonoBehaviour
+{
+    public GameObject confirmPanel;
+    public Text confirmText;
+
+    private CarList.Car _carToPurchaseObj;
+    private readonly Color32 _lightGreyColor = new Color32(147, 147, 147, 255);
+    
+    private void OnEnable()
+    {
+        RefreshPage();
+    }
+
+    // refresh the page
+    public void RefreshPage()
+    {
+        confirmPanel.SetActive(false);
+        CheckCarOwnerships();
+        SetCoinItemPrices();
+    }
+
+    // Apply discounts on coin text for each car sales in coin
+    private void SetCoinItemPrices()
+    {
+        var discount = GameDataController.GetGameData().Discount;
+        foreach (var car in CarList.List.Where(car => !car.IsPriceInDollar))
+        {
+            car.StoreItemCarGameObj.transform.Find("price/priceValue").gameObject.GetComponent<Text>().text =
+                (car.Price * discount).ToString(CultureInfo.InvariantCulture);
+        }
+    }
+    
+    public void OnItemSedanClicked()
+    {
+        _carToPurchaseObj = CarList.CarSedan;
+        BuyCars();
+    }
+
+    public void OnItemTruckClicked()
+    {
+        var gameData = GameDataController.GetGameData();
+        // players can purchase the coin item only it they have enough coin
+        if (gameData.coinsOwned >= CarList.CarTruck.Price * gameData.Discount)
```

Comment:
disable truck if user doesnt have enough coins? or pop a dialog to let the user know? can be a TODO

Reply:
yea you are right, I should have something to reminder the user here. It will be a todo.